applications/StructuralMechanicsApplication/custom_advanced_constitutive/plane_stress_d_plus_d_minus_damage_masonry_2d.cpp

```diff
@@ -7,752 +7,1076 @@
-bool DamageDPlusDMinusMasonry2DLaw::Has(const Variable<Matrix>& rThisVariable)
+/***********************************************************************************/
+void DamageDPlusDMinusMasonry2DLaw::CalculateMaterialResponsePK2 (
+	Parameters& rValues)
 {
-    if(rThisVariable == INTEGRATED_STRESS_TENSOR){
-        // explicitly returning "false", so we know we must call CalculateValue(...)
-        return false;
-    }
-    return BaseType::Has(rThisVariable);
+	CalculateMaterialResponseCauchy(rValues);
 }
 /***********************************************************************************/
 /***********************************************************************************/
-void DamageDPlusDMinusMasonry2DLaw::SetValue(
-    const Variable<double>& rThisVariable,
-    const double& rValue,
-    const ProcessInfo& rCurrentProcessInfo
-    )
-{
-    if (rThisVariable == DAMAGE_TENSION) {
-       mTensionDamage = rValue;
-    } else if (rThisVariable == THRESHOLD_TENSION) {
-       mTensionThreshold = rValue;
-    } else if (rThisVariable == DAMAGE_COMPRESSION) {
-       mCompressionDamage = rValue;
-    } else if (rThisVariable == THRESHOLD_COMPRESSION) {
-       mCompressionThreshold = rValue;
-    } else if (rThisVariable == UNIAXIAL_STRESS_COMPRESSION) {
-       mCompressionUniaxialStress = rValue;
-    } else if (rThisVariable == UNIAXIAL_STRESS_TENSION) {
-       mTensionUniaxialStress = rValue;
-    } else {
-       return BaseType::SetValue(rThisVariable, rValue, rCurrentProcessInfo);
-    }
+void DamageDPlusDMinusMasonry2DLaw::CalculateMaterialResponseKirchhoff (
+	Parameters& rValues)
+{
+	CalculateMaterialResponseCauchy(rValues);
 }
 /***********************************************************************************/
 /***********************************************************************************/
-double& DamageDPlusDMinusMasonry2DLaw::GetValue(
-    const Variable<double>& rThisVariable,
-    double& rValue
-    )
-{
-    if (rThisVariable == DAMAGE_TENSION) {
-       rValue = mTensionDamage;
-    } else if (rThisVariable == THRESHOLD_TENSION) {
-       rValue = mTensionThreshold;
-    } else if (rThisVariable == DAMAGE_COMPRESSION) {
-       rValue = mCompressionDamage;
-    } else if (rThisVariable == THRESHOLD_COMPRESSION) {
-       rValue = mCompressionThreshold;
-    } else if (rThisVariable == UNIAXIAL_STRESS_COMPRESSION) {
-       rValue = mCompressionUniaxialStress;
-    } else if (rThisVariable == UNIAXIAL_STRESS_TENSION) {
-       rValue = mTensionUniaxialStress;
-    } else {
-       return BaseType::GetValue(rThisVariable, rValue);
-    }
-    return rValue;
+void DamageDPlusDMinusMasonry2DLaw::CalculateMaterialResponseCauchy (
+	Parameters& rValues)
+{
+	const ProcessInfo&  pinfo = rValues.GetProcessInfo();
+	const GeometryType& geom  = rValues.GetElementGeometry();
+	const Properties&   props = rValues.GetMaterialProperties();

```

Comment:
The style guide is not respected but is not important

Reply:
ok